src/v/storage/tests/log_segment_appender_test.cc

```diff
@@ -217,3 +218,25 @@ SEASTAR_THREAD_TEST_CASE(test_can_append_little_data) {
     BOOST_REQUIRE_EQUAL(appender.file_byte_offset(), data.size());
     appender.close().get();
 }
+
+SEASTAR_THREAD_TEST_CASE(test_full_disk_append_failure) {
+    auto f = ss::file(
```

Comment:
wondered if you considered the alternative design to have a new `file` wrapper. i.e.: `redpanda::file` class/object. that can get you a `ss::file && file() && { return std::move(_file);}` kind of method.

i didn't see an RFC attached to this PR to understand the design.

Reply:
I like this idea.  `ss::file` doesn't have the API we need, so we end up extending `ss::file_impl`, which is a bit of layering smell. Having our own filehandle object would be cleaner. Runtime bloat is not a concern if we're always going to instantiate this probe API shim always.
In terms of design, this is based on the file_sanitizer class as an example. I'll chat with @dotnwat and see if we might want to create a proper `redpanda::file` interface in the near future.